ros2pkg/ros2pkg/build_type/__init__.py

```diff
@@ -0,0 +1,160 @@
+from ros2cli.plugin_system import instantiate_extensions
+from ros2cli.plugin_system import PLUGIN_SYSTEM_VERSION
+from ros2cli.plugin_system import satisfies_version
+
+
+class BuildTypeExtension():
+    """
+    The interface for build-type extensions.
+
+    The following properties must be defined:
+    * `NAME` (will be set to the entry point name)
+
+    The following methods must be defined:
+    * create_package
+    """
+
+    NAME = None
+    EXTENSION_POINT_NAME = 'ros2pkg.build_type'
+    EXTENSION_POINT_VERSION = '0.1'
+
+    def __init__(self):
+        super(BuildTypeExtension, self).__init__()
+        satisfies_version(PLUGIN_SYSTEM_VERSION, '^0.1')
+
+    def create_package(self, args):
+        """Create the package directory structure and content including a package.xml file."""
+        raise NotImplementedError
+
+
+class PkgCreateArgs():
+    """
+    Wrapper for commandline arguments of a package-create action.
+
+    :ivar __args: A argparse.Namespace object.
+    """
+
+    def __init__(self, args):
+        self.__args = args
+
+    @property
+    def args(self):
+        return self.__args
+
+    @property
+    def build_type(self):
+        """Return build-type string."""
+        return self.args.build_type
+
+    @property
+    def dependencies(self):
+        """Return dependencies list."""
+        return self.args.dependencies
+
+    @property
+    def description(self):
+        """Return package description string."""
+        return self.args.description
+
+    @property
+    def destination_directory(self):
+        """Return the directory name (string) of new package."""
+        return self.args.destination_directory
+
+    @property
+    def library_name(self):
+        """Return package library name, may be None."""
+        return self.args.library_name
+
+    # renamed to _license due to A003 "license" is a python builtin,
+    # consider renaming the class attribute
+    @property
+    def _license(self):
+        """Return the package license string."""
+        return self.args.license
+
+    @property
+    def maintainer_email(self):
+        """Return the package mainainer email address."""
+        return self.args.maintainer_email
+
+    @property
+    def maintainer_name(self):
+        """Return the package maintainer name."""
+        return self.args.maintainer_name
+
+    @property
+    def node_name(self):
+        """Return the package node name, may be None."""
+        return self.args.node_name
+
+    @property
+    def package_format(self):
+        """Return the package.xml format number (2 or 3)."""
+        return self.args.package_format
+
+    @property
+    def package_name(self):
+        """Return the package name string."""
+        return self.args.package_name
+
+
+def get_build_type_names():
+    """
+    Get the build-type names registered with the build-type entry-point.
+
+    :return: List of build-type string names.
+    """
+    names = []
```

Comment:
Is it possible to have duplicate build type names and would it matter? maybe consider `set` vs.`list`

Reply:
No there should not be duplicates build-types even if there were duplicate entry points registered. This is because get_entry_points() returns a dictionary (build_type_name->build_type_instance). Thus, build-type names will be unique.